Std/Data/Nat/Lemmas.lean

```diff
@@ -10,7 +10,109 @@ import Std.Data.Nat.Basic
 
 namespace Nat
 
-/-! ## le/lt -/
+/-! ### rec/cases -/
+
+section recAux
+variable {motive : Nat → Sort _} (zero : motive 0) (succ : ∀ n, motive n → motive (n+1))
```

Comment:
`variable` should be avoided unless absolutely necessary, especially if you start shadowing them like below. For these theorems you can use either autoimplicits or `(zero succ)` binders in each theorem.

Reply:
`elab_as_elim` got in the way of using autoimplicits but I did remove all `variable` commands.